iothub/service/src/Registry/Models/Device.cs

```diff
@@ -13,13 +13,6 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public class Device : IETagHolder
     {
-        /// <summary>
-        /// Creates a new instance of <see cref="Device"/>
-        /// </summary>
-        public Device()
```

Comment:
Isn't this needed for deserialization?

Reply:
Users should not be constructing Devices without Ids, so they shouldn't be able to access this constructor. I'll add this back in but make it internal.